File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    for(int i= 0; i<commands.size(); i++){
        vector<int> v;
        int x = commands[i][0] - 1;
        int y = commands[i][1] - 1;
        int z = commands[i][2] - 1;

        for(int j = x; j<=y; j++){
            v.push_back(array[j]);
        }
        sort(v.begin(), v.end());

        answer.push_back(v[z]);
    }
    
    
    return answer;
}

File: 프로그래머스/1/42748. K번째수/K번째수.js
function solution(array, commands) {
    var answer = [];
    
    for(let c = 0; c<commands.length; c++){
        let newArray = [];

        for(let i = commands[c][0]-1; i<= commands[c][1]-1; i++){
            newArray.push(array[i]);
        }

        newArray.sort((a, b) => {if(a > b) return 1; else if(a < b) return -1; else return 0;})

        answer.push(newArray[commands[c][2]-1]);
    }
    return answer;
}